src/XRP/serializer.ts

```diff
@@ -135,6 +156,43 @@ const serializer = {
   xrpAmountToJSON(xrpDropsAmount: XRPDropsAmount): string {
     return `${xrpDropsAmount.getDrops()}`
   },
+
+  /**
+   * Convert an array of Memo objects to a JSON representation keyed by
+   * a field called 'Memos' iff the array is not empty and it contains
+   * non-empty objects.
+   *
+   * @param memos
+   * @returns the Memos as JSON or undefined
+   */
+  memosToJSON(memos: Memo[]): { Memos: MemoJSON[] } | undefined {
+    if (!memos || !memos.length) {
+      return undefined
+    }
+
+    const filteredMemos = memos
+      .filter((memo) => !!memo)
+      .map((memo) => this.memoToJSON(memo))
+    return (filteredMemos.length && { Memos: filteredMemos }) || undefined
+  },
+
+  /**
+   * Convert a Memo to a JSON representation.
+   *
+   * @param memo - The Memo to convert.
+   * @returns The Memo as JSON.
+   */
+  memoToJSON(memo: Memo): MemoJSON {
+    const jsonMemo: MemoDetailsJSON = {
+      MemoData: memo?.getMemoData()?.getValue_asB64(),
+      MemoFormat: memo?.getMemoFormat()?.getValue_asB64(),
+      MemoType: memo?.getMemoType()?.getValue_asB64(),
+    }
+
+    return {
+      Memo: jsonMemo,
+    }
```

Comment:
This feels a little odd. Isn't `jsonMemo` already a memo in JSON format? Why do we need to create a wrapper JSON object with only a `Memo` key? Is the extra layer of indirection necessary?

Reply:
That's the format of the JSON that's stored in the transaction. 

https://xrpl.org/transaction-common-fields.html#memos-field

Reply:
Jeez. I guess I'm just unhappy with `rippled`.